broker/broker_schema_mutator.go

```diff
@@ -50,34 +59,73 @@ func (b *BrokerSchemaMutator) GetTable(name string) (table *common.Table, err er
 }
 
 func (b *BrokerSchemaMutator) CreateTable(table *common.Table) (err error) {
+	b.tables[table.Name] = memCom.NewTableSchema(table)
 	return
 }
 func (b *BrokerSchemaMutator) DeleteTable(name string) (err error) {
+	delete(b.tables, name)
 	return
 }
 func (b *BrokerSchemaMutator) UpdateTableConfig(table string, config common.TableConfig) (err error) {
+	b.tables[table].Schema.Config = config
 	return
 }
 func (b *BrokerSchemaMutator) UpdateTable(table common.Table) (err error) {
+	b.tables[table.Name] = memCom.NewTableSchema(&table)
 	return
 }
 func (b *BrokerSchemaMutator) AddColumn(table string, column common.Column, appendToArchivingSortOrder bool) (err error) {
+	oldSchema := b.tables[table].Schema
+	oldSchema.Columns = append(oldSchema.Columns, column)
+	if appendToArchivingSortOrder {
+		oldSchema.ArchivingSortColumns = append(oldSchema.ArchivingSortColumns, len(oldSchema.Columns)-1)
+	}
+	b.tables[table] = memCom.NewTableSchema(&oldSchema)
 	return
 }
 func (b *BrokerSchemaMutator) UpdateColumn(table string, column string, config common.ColumnConfig) (err error) {
+	oldSchema := b.tables[table].Schema
+	target := -1
+	for i, col := range oldSchema.Columns {
+		if col.Name == column {
+			target = i
+			break
+		}
+	}
+	if target == -1 {
+		err = errors.New(fmt.Sprintf("column %s not found", column))
+		return
+	}
+	oldSchema.Columns[target].Config = config
+	b.tables[table] = memCom.NewTableSchema(&oldSchema)
 	return
 }
 
 func (b *BrokerSchemaMutator) DeleteColumn(table string, column string) (err error) {
+	oldSchema := b.tables[table].Schema
+	target := -1
+	for i, col := range oldSchema.Columns {
+		if col.Name == column {
+			target = i
+			break
+		}
+	}
+	if target == -1 {
+		err = errors.New(fmt.Sprintf("column %s not found", column))
+		return
+	}
+	oldSchema.Columns[target].Deleted = true
```

Comment:
why not change the value in place? Seems like it's a pointer

Reply:
you can avoid allocate a new object via this way

Reply:
because deleting a table will also cause `ValueTypeByColumn` to change, I didn't want to duplicate the logic in `NewTableSchema` here. better recreate by calling `NewTableSchema` since it's not a frequent call.